Update package docs with all configurable fields (#2031)
crossplane/crossplane: docs/introduction/packages.md

> may be installed into the cluster. They are strictly used as metadata in a
> Crossplane package.

A Provider package may optionally contain one or more CRDs. These CRDs will be
installed prior to the creation of the Provider's `Deployment`. Crossplane will
not install _any_ CRDs for a package unless it can determine that _all_ CRDs can
be installed. This guards against multiple Providers attempting to reconcile the
same CRDs. Crossplane will also create a `ServiceAccount` with permissions to
reconcile these CRDs and it will be assigned to the controller `Deployment`.

The `spec.controller.image` fields specifies that the `Provider` desires for the
controller `Deployment` to be created with the provided image. It is important
to note that this image is separate from the package image itself. In the case
above, it is an image containing the `provider-gcp` controller binary.

The `spec.controller.permissionRequests` field allows a package author to
request additional RBAC for the packaged controller. The controller's
`ServiceAccount` will automatically give the controller permission to reconcile
all types that its package installs, as well as `Secrets`, `ConfigMaps`, and

> `Events`. Any additional permissions must be explicitly requested.

Reviewer: Thanks for documenting this!

> > Note that the Crossplane RBAC manager can be configured to reject permissions
> > for certain API groups. If a package requests permissions that Crossplane is
> > configured to reject, the package will fail to be installed. 

Reviewer: ```suggestion
> Note that the Crossplane RBAC manager must be configured to allow permissions
> for certain API groups. If a package requests permissions that Crossplane is not
> configured to allow, the package will fail to be installed. 
```

Reviewer: Just flipping this phrasing since we require the user to pass a `ClusterRole` of permissions we allow, rather than permissions we reject.

Author: Ah great point! Thanks :)
